ext/reflection/php_reflection.c

```diff
@@ -305,6 +313,8 @@ static void reflection_free_objects_storage(zend_object *object) /* {{{ */
 		case REF_TYPE_PARAMETER:
 			reference = (parameter_reference*)intern->ptr;
 			_free_function(reference->fptr);
+			/* fallthrough */
+		case REF_TYPE_TYPE:
```

Comment:
@nikic I'm afraid I'll have to leave further tests about `_copy_function` / `_free_function` to you, or I will need some pointers.

I've also added tests for ReflectionFunction($closure) and ReflectionMethod($closure, '__invoke') and the latter doesn't seem to work with return types. I'll look into it as soon as I have time.